streaming/base/storage/upload.py

```diff
@@ -945,6 +947,137 @@ def check_folder_exists(self):
+        if encrypt_type:
+            if getattr(EncryptType, encrypt_type, None) is None:
+                raise ValueError(
+                    f'Invalid ALIPAN_ENCRYPT_TYPE: {encrypt_type}. Encryption type must be one of `Simple`, `ChaCha20`, `AES256CBC`'
+                )
+            else:
+                self.alipan_encrypt_type = getattr(EncryptType, encrypt_type)
+
+        self.api = AliPCSApiMix(web_refresh_token, web_token_type=web_token_type)
+        self.check_token()
+
+    def check_token(self):
+        """Raise an exception if the refresh_token is invalid.
+
+        Raises:
+            error: AliPCSError with code `AccessTokenInvalid`
+        """
+        self.api.refresh()
+
+    def upload_file(self, filename: str):
+        """Upload file from local instance to Alipan directory.
+
+        Args:
+            filename (str): File to upload.
+        """
+        from alipcs_py.commands import upload as alipcs_upload
+        from alipcs_py.commands.upload import EncryptType, total_len
+
+        @retry(num_attempts=self.retry)
+        def _upload_file():
+            local_filename = os.path.join(self.local, filename)
+            remote_filename = os.path.join(self.remote, filename)  # pyright: ignore
+            obj = urllib.parse.urlparse(remote_filename)
+            logger.debug(f'Uploading to {remote_filename}')
+            if self.alipan_encrypt_type == EncryptType.No:
+                file_size = os.stat(local_filename).st_size
+            else:
+                file_size = self.alipan_encrypt_type
+                encrypt_io = self.alipan_encrypt_type.encrypt_io(open(local_filename, 'rb'),
+                                                                 self.alipan_encrypt_password)
+                file_size = total_len(encrypt_io)
+            with tqdm.tqdm(total=file_size,
+                           unit='B',
+                           unit_scale=True,
+                           desc=f'Uploading to {remote_filename}',
+                           disable=(not self.progress_bar)) as pbar:
+                alipcs_upload.upload_file(
+                    self.api, (local_filename, obj.path),
+                    'overwrite',
+                    encrypt_password=self.alipan_encrypt_password,
+                    encrypt_type=self.alipan_encrypt_type,
+                    callback_for_monitor=lambda offset: pbar.update(offset - pbar.n))
+            self.clear_local(local=local_filename)
+
+        _upload_file()
+
+    def list_objects(self, prefix: Optional[str] = None) -> List[str]:
+        """List all objects in the remote path with the given prefix.
+
+        Args:
+            prefix (Optional[str], optional): The prefix to search for. Defaults to ``None``.
+
+        Returns:
+            List[str]: A list of object names that match the prefix.
+        """
+        if prefix is None:
+            prefix = ''
+
+        assert self.remote is not None
```

Comment:
Please avoid `assert`, instead raise a python exception with a descriptive error message.

Reply:
I give a `ValueError` when `self.remote` is None.